### PR TITLE
Add feature to filter by field in the events table rows

### LINKED ISSUE #6977: Add feature to filter by field in the events table rows
| Wazuh | Rev  | Browser                      |
|-------|------|------------------------------|
| 4.9.0 | 07 | Chrome, Firefox, Safari, etc |

# Description

The native Discover plugins has the feature to add a filter in one of the results rows. We need to implement this feature as well in the new events table.

#### Native discover feature

image

#### Events table

image

### PR DESCRIPTION
**Pull request by guidomodarelli**
### Description
The native Discover plugins has the feature to add a filter in one of the results rows. We need to implement this feature as well in the new events table.
 
### Issues Resolved
#6977 

### Evidence

https://github.com/user-attachments/assets/ea998a25-8c4a-4742-9557-849aeba85aad

### Test
Navigate to the "Events" tab in any module, place the cursor over a cell, and click on "+" or "-" to filter by this value.

### Unit tests

image

image

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by JuanGarriuz** [comment]
### Test

image

File: plugins/main/public/components/common/data-grid/use-data-grid.ts [hunk 1]
```diff
@@ -43,10 +66,14 @@ export const useDataGrid = (props: tDataGridProps): EuiDataGridProps => {
...
     renderColumns,
-    pagination: defaultPagination,
+    useDefaultPagination = false,
+    pagination: paginationProps = {},
+    filters,
+    setFilters,
   } = props;
   /** Columns **/
-  const [columns, setColumns] = useState<tDataGridColumn[]>(defaultColumns);
+  const [columns /* , setColumns */] =
```
Comment (JuanGarriuz) [inline plugins/main/public/components/common/data-grid/use-data-grid.ts:75]: Why is it commented?

> /* , setColumns */

Comment (guidomodarelli) [reply plugins/main/public/components/common/data-grid/use-data-grid.ts:75]: Because it is not in use.

**Review by asteriscos — CHANGES_REQUESTED** [review]
Test: :red_circle: 

Please check that the buttons don't seem to work after changing the page.

Peek 2024-09-16 19-33

image

**Review by Machi3mfl — APPROVED** [review]
CR: ✅ 
Test: ✅ 

Screenshot 2024-09-16 at 16 15 51

Screenshot 2024-09-16 at 16 15 35

**Review by asteriscos — APPROVED** [review]

**Review by JuanGarriuz — APPROVED** [review]
LGTM!!

**Pull request merged**